### PR TITLE
feat(Grid): enable zero span columns

### PR DESCRIPTION
**Pull request by zachhardesty7**
Adds functionality requested in (#2435) with more convenient syntax.

Since Grid components may often have their classes added programatically, enabling the use of span "0" simplifies the code and improves readability. From an English language syntax perspective, content "spanning 0 columns" makes sense in a similar way as content with a "width of 0." Additionally, several other major libraries support this notation.

#### Changelog

**New**

- Extend SCSS to generate class `.bx--col-0` that sets `display: none` on a given column

#### Testing / Reviewing

The example docs now include a demonstration of the class in action!

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
Looks basically good, wanted to see what @joshblack thinks. Great to have some clarification of `for` loop starting `-1` (Is it some kind of Sass trick?). Thanks @zachhardesty7!

File: packages/grid/examples/preview/src/components/App.js [hunk 1]
```diff
@@ -158,6 +158,37 @@ const sections = [
       },
     ],
   },
+  {
+    name: 'Hide column per breakpoint',
+    description:
+      'Use the column helpers to specify a column span of 0 at different widths',
+    content: () => (
```
Comment (joshblack) [inline packages/grid/examples/preview/src/components/App.js]: I _think_ this isn't mirroring the `{ name, description, examples }` structure used by the other examples which is causing the preview for this example to error out. Mind updating it real quick to match the other structures so we can render the preview? 👀

**Review by joshblack — CHANGES_REQUESTED** [review]

**Comment by zachhardesty7** [comment]
@asudoh Honestly, I wrote this quite a while ago as a feature of the (upcoming) React Grid components. I have no idea why I put `-1` instead of `0`. I tested it with both and `-1` indeed allows an incorrect span. I fixed it here: 7354c74

File: packages/grid/examples/preview/src/components/App.js (hunk 1 above)
Comment (zachhardesty7) [reply packages/grid/examples/preview/src/components/App.js]: I think I messed up and copied it into the wrong location. I meant to have it as part of the "Usage" section. The similar structure of nested objects probably tripped me up. Fixed it here: 267b28f

File: packages/grid/scss/_mixins.scss [hunk 2]
```diff
@@ -320,9 +327,12 @@ $carbon--aspect-ratios: ((16, 9), (2, 1), (4, 3), (1, 1), (1, 2));
     @include carbon--make-row();
   }
 
-  .#{$prefix}--grid--condensed [class*='#{$prefix}--col'] {
-    padding-top: $condensed-gutter / 2;
-    padding-bottom: $condensed-gutter / 2;
+  .#{$prefix}--grid--condensed .#{$prefix}--row:not(:last-of-type) {
```
Comment (joshblack) [inline packages/grid/scss/_mixins.scss]: I think we'll want to keep the padding changes from: https://github.com/carbon-design-system/carbon/pull/4960 👍

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/grid/scss/_mixins.scss (hunk 2 above)
Comment (zachhardesty7) [reply packages/grid/scss/_mixins.scss]: Not sure how that happened, I must have merged something incorrectly. Git can be confusing haha, a778420

Comment (joshblack) [reply packages/grid/scss/_mixins.scss]: Totally get that 😂 😂

**Review by joshblack — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]
I think LGTM.

**Review by abbeyhrt — APPROVED** [review]
Looks good to me! Thank you for doing this!

**Pull request merged**